### PR TITLE
Improvements to isometric tilemaps

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

Upon working a bit on heeve I thought of a couple improvements to the tile map api to make it clearer.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `flutter format` and the `flutter analyze` does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

**Review by wolfenrain — APPROVED** [review]

File: packages/flame/lib/src/components/isometric_tile_map_component.dart [hunk 1]
```diff
@@ -72,26 +86,39 @@ class IsometricTileMapComponent extends PositionComponent {
...
-          final p = getBlockPositionInts(j, i);
+          final p = getBlockRenderPositionInts(j, i);
           sprite.render(c, position: p, size: size);
         }
       }
     }
   }
 
-  /// Get the position in which a block must be in the isometric space.
+  /// Get the position in which a block must be rendered in the isometric space.
```
Comment (spydon) [inline packages/flame/lib/src/components/isometric_tile_map_component.dart:96]:
```suggestion
  /// Get the position in which a block is rendered in, in the isometric space.
```

**Review by spydon — APPROVED** [review]
Lgtm!
Maybe we should also get rid of all the `get`-prefixes before v1?
https://dart.dev/guides/language/effective-dart/design#avoid-starting-a-method-name-with-get

File: packages/flame/lib/src/components/isometric_tile_map_component.dart [hunk 2]
```diff
@@ -8,14 +10,26 @@ import 'position_component.dart';
...
+  @override
+  bool operator ==(Object other) {
+    if (other is! Block) {
+      return false;
+    }
+    return other.x == x && other.y == y;
+  }
+
+  @override
+  int get hashCode => hashValues(x, y);
```
Comment (erickzanardo) [inline packages/flame/lib/src/components/isometric_tile_map_component.dart:32]: A toVector method would be interesting too, what do you think?

**Review by erickzanardo — APPROVED** [review]

**Comment by luanpotter** [comment]
@spydon I think we can discuss that in a followup, I am start to have reservations with this dart get hatred :P

**Comment by spydon** [comment]
> @spydon I think we can discuss that in a followup, I am starting to have reservations with this dart get hatred :P

For sure, wasn't meant for this PR, just something we could refactor in the future. :)

File: packages/flame/lib/src/components/isometric_tile_map_component.dart (hunk 2 above)
Comment (erickzanardo) [reply packages/flame/lib/src/components/isometric_tile_map_component.dart:32]: Yeah, just using the x, y from the block to create a vector2, we had to that manually on heeve to make the unit movement, so I think could be a good shortcut.

File: packages/flame/lib/src/components/isometric_tile_map_component.dart [hunk 3]
```diff
@@ -8,14 +10,26 @@ import 'position_component.dart';
...
   /// x and y coordinates on the matrix
-  int x, y;
+  final int x, y;
 
-  Block(this.x, this.y);
+  const Block(this.x, this.y);
 
   @override
   String toString() => '($x, $y)';
+
```
Comment (spydon) [inline packages/flame/lib/src/components/isometric_tile_map_component.dart:22]:
```suggestion

Vector2 toVector2() => Vector2.fromInts(x, y);

```

**Pull request merged**